### ISSUE TITLE
Test.hs stalls with amdgpu + NixOS 20.03

### ISSUE THREAD

**Issue by NinjaTrappeur** [guibou/nixGL#36]
Running `./Test.hs` end up stalling on my system. It's been displaying 

```
It can take a while, this will build and test all drivers in the background
```

for more than 45 minutes. No CPU activity according to htop, no network activity according to nethogs.

System:
- driver: amdgpu
- GPU: radeon 590
- system: NixOS 20.03

I can provide you with a live tmate session to debug that if needed.

**Comment by guibou** [comment]
@NinjaTrappeur confirmed that the test are actually working. I don't understand why it stalled.

In the meantime, #37 is an attempt at fixing @NinjaTrappeur configuration.

**Comment by NinjaTrappeur** [comment]
I did not manage to reproduce the stall, even with a brand new repo
checkout.

Could we have some leaking state elsewhere?

**Comment by guibou** [comment]
I don't know. `Test.hs` is a bit weird. It runs many sub process, calling `nix-shell` and this is not even a real (i.e. compilled) haskell program : it is run in interpreted mode, which may be responsible for dead lock.

**Comment by guibou** [comment]
I'm closing this. There may be a lot of think to imporve the test, but #39 at least should have fixed the stale and #37 fixs the bug with the mesa driver with radeon card.
